File: push/src/main/java/com/xiaomi/push/service/PushHostManagerFactoryAspect.java
package com.xiaomi.push.service;

import android.net.Uri;

import org.aspectj.lang.ProceedingJoinPoint;
import org.aspectj.lang.annotation.Around;
import org.aspectj.lang.annotation.Aspect;

@Aspect
public class PushHostManagerFactoryAspect {

    @Around("execution(* com.xiaomi.push.service.PushHostManagerFactory.GslbHttpGet.doGet(..)) && args(url)")
    public Object addCountryCodeToConfigurationUrl(final ProceedingJoinPoint joinPoint, String url) throws Throwable {
        url = url.replaceFirst("&countrycode=[^&]+", "");
        url = url.replaceFirst("\\?countrycode=[^&]+", "?");
        Uri.Builder uri = Uri.parse(url).buildUpon();
        uri.appendQueryParameter("countrycode", "CN");
        return joinPoint.proceed(new Object[]{uri.toString()});
    }
}

File: mipush_hook/src/main/java/com/xiaomi/push/service/PushHostManagerFactoryAspect.kt
package com.xiaomi.push.service

import android.net.Uri
import org.aspectj.lang.ProceedingJoinPoint
import org.aspectj.lang.annotation.Around
import org.aspectj.lang.annotation.Aspect

@Aspect
class PushHostManagerFactoryAspect {
    @Around("execution(* com.xiaomi.push.service.PushHostManagerFactory.GslbHttpGet.doGet(..)) && args(url)")
    @Throws(Throwable::class)
    fun addCountryCodeToConfigurationUrl(joinPoint: ProceedingJoinPoint, url: String): Any? {
        var newUrl = url
        newUrl = newUrl.replaceFirst("&countrycode=[^&]+".toRegex(), "")
        newUrl = newUrl.replaceFirst("\\?countrycode=[^&]+".toRegex(), "?")
        val uri = Uri.parse(newUrl).buildUpon()
        uri.appendQueryParameter("countrycode", "CN")
        return joinPoint.proceed(arrayOf(uri.toString()))
    }
}
